### PR TITLE
Use DayNight themes

### LINKED ISSUE #873: Add custom theme
Add an additional theme where primary and accent color can be selected via a color picker.

### LINKED ISSUE #1058: "Info ssl client cert" tooltip has bright background in dark themes
### Actual behaviour
"Info ssl client cert" tooltip has bright background in dark themes, which makes the tooltip hard to read.

### Expected behaviour
"Info ssl client cert" tooltip should have a dark background.

#### Client
* Android version: 5.1.1

### PR DESCRIPTION
**Pull request by mueller-ma**
Fixes #873
Fixes #1058
Closes #1431

TODO:

- [x] Fix crash when opening About on Android 5
- [x] Fix section switch widgets

### CODE REVIEW

**Comment by mueller-ma** [comment]
[mobile-foss-beta-debug.zip](https://github.com/openhab/openhab-android/files/3649074/mobile-foss-beta-debug.zip)

**Comment by mueller-ma** [comment]
> We can/should override this in widgetlist_sectionswitchitem_button.xml.

:+1:

> I'm still not really convinced that the accent color selection is a useful addition (for the same reasons as mentioned earlier, but it looks like it's important to you?

It's possible to change the accent color with the current implementation and it's also possible to the Basic UI and many other UIs. Thus I don't want to remove this feature from the app.

**Comment by maniac103** [comment]
> It's possible to change the accent color with the current implementation 

Yes, but only between a set of color schemes *related to openHAB*.

> and it's also possible to the Basic UI and many other UIs. 

Where can one change the Basic UI color scheme? I've only ever seen such settings in some Android apps not affiliated to a certain project or company so far.

> Thus I don't want to remove this feature from the app.

You want to add this feature; that's not the same thing as not removing it. One currently can't select arbitrary color schemes.

**Comment by mueller-ma** [comment]
> Where can one change the Basic UI color scheme? I've only ever seen such settings in some Android apps not affiliated to a certain project or company so far.

I've misread a PR for theming Basic UI: https://github.com/eclipse/smarthome/pull/4137
The colors didn't make it through review, only bright and dark mode. I reduced the accent colors to orange, blue (like Basic UI) and grey (like the old grey theme).

**Comment by maniac103** [comment]
> I've misread a PR for theming Basic UI: eclipse/smarthome#4137
> The colors didn't make it through review, only bright and dark mode.

Heh, I see Kai shared my concerns in https://github.com/eclipse/smarthome/pull/4137/files#r137015539 ;-)

> I reduced the accent colors to orange, blue (like Basic UI) and grey (like the old grey theme).

Nice. Now we just need to restore the pink accent color in the blue theme to make it consistent to BasicUI and we're set, I think.

File: mobile/src/main/res/layout/item_updating_pref_dialog.xml [hunk 1]
```diff
@@ -33,6 +33,7 @@
```
[the remark this answers is not in the document]
Comment (maniac103) [reply mobile/src/main/res/layout/item_updating_pref_dialog.xml]: For your consideration: https://github.com/maniac103/openhab.android/commit/a2629b0a97eddf9fd91ed285da86f72ebec98d11

File: mobile/src/androidTest/java/org/openhab/habdroid/ui/WidgetAdapterTest.kt [hunk 2]
```diff
@@ -26,7 +27,8 @@ class WidgetAdapterTest {
 
     @Test
     fun testColorMappingDarkTheme() {
-        val colorMapper = ColorMapper(ContextThemeWrapper(context, R.style.HABDroid_Basic_ui_dark))
+        AppCompatDelegate.setDefaultNightMode(AppCompatDelegate.MODE_NIGHT_NO)
```
[the remark this answers is not in the document]
Comment (mueller-ma) [reply mobile/src/androidTest/java/org/openhab/habdroid/ui/WidgetAdapterTest.kt]: > either there's a mixup here

Yes, there is. However `setDefaultNightMode()` doesn't make the theme dark here. I added another theme for this.

[the remark this answers is not in the document]
Comment (maniac103) [reply mobile/src/androidTest/java/org/openhab/habdroid/ui/WidgetAdapterTest.kt]: I'd suggest a different approach for this, inspired by `AppCompatDelegateImpl`'s implementation: https://github.com/maniac103/openhab.android/commit/4f8aba8dc7fd10865befedf5e56912b544e25c7a

**Comment by mueller-ma** [comment]
> I also wonder whether we shouldn't go to minSdkVersion 21

I'd like to keep this for now.

File: mobile/src/main/java/org/openhab/habdroid/ui/preference/ItemUpdatingPreference.kt [hunk 3]
```diff
@@ -174,17 +174,18 @@ class ItemUpdatingPreference constructor(context: Context, attrs: AttributeSet?)
...
+                editorWrapper.error = null
             }
             updateOkButtonState()
         }
 
         private fun updateOkButtonState() {
             val dialog = this.dialog
             if (dialog is AlertDialog) {
-                dialog.getButton(AlertDialog.BUTTON_POSITIVE).isEnabled = !editor.isEnabled || editor.error == null
+                dialog.getButton(AlertDialog.BUTTON_POSITIVE).isEnabled = !editor.isEnabled ||
```
Comment (maniac103) [inline mobile/src/main/java/org/openhab/habdroid/ui/preference/ItemUpdatingPreference.kt]: Total nit: maybe put whole assignment (everything after `=`) to new line?

**Comment by maniac103** [comment]
Hmm, opening settings has a delay of at least half a second on a fairly fast device (OnePlus 6). This seems too much for my taste, and I don't think it's always been like that, right?

File: mobile/src/main/java/org/openhab/habdroid/ui/AbstractBaseActivity.kt [hunk 4]
```diff
@@ -47,6 +49,13 @@ abstract class AbstractBaseActivity : AppCompatActivity(), CoroutineScope {
     @CallSuper
     override fun onCreate(savedInstanceState: Bundle?) {
         setTheme(Util.getActivityThemeId(this))
+        val wantedNightMode = getPrefs().getDayNightMode(this)
+        val currentNightMode = AppCompatDelegate.getDefaultNightMode()
+
+        if (wantedNightMode != currentNightMode) {
+            AppCompatDelegate.setDefaultNightMode(wantedNightMode)
+            recreate()
```
Comment (maniac103) [inline mobile/src/main/java/org/openhab/habdroid/ui/AbstractBaseActivity.kt]: Why is this call needed? At this point, no resources are inflated yet...

**Comment by maniac103** [comment]
Got 2 additional optimization commits to pull: https://github.com/maniac103/openhab.android/commits/dark-mode

**Pull request merged**